client/state/products-list/selectors/compute-full-and-monthly-prices-for-plan.js

```diff
@@ -30,9 +31,16 @@ export const computeFullAndMonthlyPricesForPlan = (
 		? getProductCost( state, planObject.getStoreSlug() )
 		: getPlanPrice( state, siteId, planObject, false );
 
+	const isIntroductoryOfferApplied = isIntroductoryOfferAppliedToPlanPrice(
+		state,
+		siteId,
+		planObject.getStoreSlug()
```

Comment:
It's not completely clear to me what arguments are optional in `computeFullAndMonthlyPricesForPlan` but I see that in at least one place above, `planObject` is allowed to be undefined (`planObject?.group`). If there's any chance of that, we might want to add optional chaining here too (or maybe just not call `isIntroductoryOfferAppliedToPlanPrice` if `planObject` does not exist). If you're confident that it always exists, though, then feel free to ignore this.

Reply:
`planObject` is being used in other places, not only here, so I wonder if there was a misunderstanding when such check was introduced.

@cbauerman The `planObject?.group` optional chaining was added in #57035 Do you recall why it was needed? It doesn't seem to do much, given that `planObject.getProductId()` is called right above, in the previous statement.

Reply:
I followed up on this and decided to remove the optional chaining for the reasons described in my previous comments. It's still working as expected.

Reply:
I can confirm that it was definitely a misunderstanding on my part. Thanks for catching it!